### DISCUSSION TITLE
五笔拼音混合输入时能否实现第一码以五笔为主，不显示拼音

### DISCUSSION THREAD

**Discussion by lgq518** [rime/home#1733] [General]
五笔拼音混合输入时能否实现第一码以五笔为主，不显示拼音，或者拼音往后排，用五笔拼音混输入多数是遇到不会拆的字才用到拼音，极少第一码就用到拼音的，第一码以五笔为主，不显示拼音，更好一些。
![截图1](https://github.com/user-attachments/assets/cbb856d2-b3ad-4f1d-8746-234e49b934a3)
输入首码后从第2候选字以后都是拼音，能不能像微软五笔混合输入一样，第一码主要为五笔。
![截图2](https://github.com/user-attachments/assets/15135213-fb6a-40a1-9b2d-9454fedffe08)

**Comment by lotem** [comment]
但是第一碼除了一級簡碼，五筆也沒有其他候選字了吧。

如果要規則統一（不計第幾碼），那麼拼音候選應該要優先於沒輸全的五筆候選，不然大多數拼音候選都見不到了。

一定要給第一碼定製成不顯示拼音的話，可知拼音 `a`、`o`、`e` 就沒法打出了。這肯定得用家做主，自己定義。

可利用拼寫運算將反查用的拼音方案修改爲沒有一碼的拼音；取消“音序”簡拼，同時給單字母的拼音設計一個反查用的特殊拼法，比如雙寫。

**Reply by lgq518** [reply]
回复的真快，辛苦了，怎样修改拼音方案修改沒有一碼的拼音？

**Reply by lotem** [reply]
五筆用的反查詞典是簡體的拼音：
https://github.com/rime/rime-pinyin-simp 

定義單字母簡拼的是這麼一行：
https://github.com/rime/rime-pinyin-simp/blob/c0b1cc5553caed77e1338adc08d6dc048c50c813/pinyin_simp.schema.yaml#L53

不加這行配置就沒有單字母的簡拼了。然而原本是一個字母的全拼還有。

**Reply by lgq518** [reply]
这一行我已经删掉了，删掉以后对简拼有用，没有简拼了，一码还是出拼音，没有改变。

**Reply by yrljroli** [reply]
可能尝试将反查的拼音词库中所有一码拼音删掉。

以<u>朙月拼音</u>为例。删除所有``.	.\r\n``即可。相应的、如果使用该拼音方案打字、将同样不能打出任何一码字。

可能的解决方案。

* 复制一份专门的拼音反查方案和词库。
* 删除词库中所有注释、将所有``(.	.\r\n)``替换为``# \1``、须要使用拼音时替换回来。
* 不再使用拼音。
